resqpy/well.py

```diff
@@ -317,222 +316,248 @@ class DeviationSurvey():
-         load... methods which offer more control arguments;
+         angle_uom (string): a resqml angle unit; should be 'dega' or 'rad'  # TODO: check this
+         measured_depths (np.array): 1d array
+         azimuths (np.array): 1d array
+         inclindations (np.array): 1d array
+         station_count (int): length of measured_depths, azimuths & inclinations
+         first_station (tuple): (x, y, z) of first point in survey, in crs for md datum
+         is_final (bool): whether survey is a finalised deviation survey
+
+      Returns:
+         DeviationSurvey
+
+      Notes:
          this method does not create an xml node, nor write hdf5 arrays
       """
 
-      assert deviation_survey_root is None or data_frame is None
       self.model = parent_model
-      self.root_node = None
-      self.uuid = None
-      self.is_final = False         # could default to True here
-      self.station_count = 0        # length of measured_depths, azimuths & inclinations
-      self.md_uom = None
-      self.angles_in_degrees = None # boolean: True for degrees, False for radians (nothing else supported)
-      self.measured_depths = None   # 1d numpy array
-      self.azimuths = None          # 1d numpy array
-      self.inclinations = None      # 1d numpy array
-      self.first_station = None     # (x, y, z) of first point in survey, in crs for md datum
+      self.is_final = is_final
+      self.md_uom = bwam.rq_length_unit(md_uom)
+
+      self.angles_in_degrees = angle_uom.strip().lower().startswith('deg')
+      """boolean: True for degrees, False for radians (nothing else supported). Should be 'dega' or 'rad'"""
+
+      # Array data
+      self.measured_depths = _as_optional_array(measured_depths)
+      self.azimuths        = _as_optional_array(azimuths)
+      self.inclinations    = _as_optional_array(inclinations)
+
+      if station_count is None and measured_depths is not None:
+         station_count = len(measured_depths)
+      self.station_count = station_count
+      self.first_station = first_station
+
       self.md_datum = md_datum      # md datum is an object in its own right, with a related crs!
       self.wellbore_interpretation = represented_interp
-      if deviation_survey_root is not None:
-         self.load_from_xml(deviation_survey_root)
-      elif data_frame is not None:
-         self.load_from_data_frame(data_frame, md_uom = length_uom, md_datum = md_datum)
-      elif deviation_survey_file:
-         self.load_from_ascii_file(deviation_survey_file,
-                                   space_separated_instead_of_csv = survey_file_space_separated,
-                                   md_uom = length_uom, md_datum = md_datum)
-      if self.uuid is None: self.uuid = bu.new_uuid()
 
+      # TODO: remove init from root_node, use just uuid
+      if root_node is not None:
+         warnings.warn("Argument root_node is deprecated, please use uuid")
```

Comment:
Does this end up as a standard python logging warning message?

Reply:
It's a standard warning, not quite the same as logging.warning.

From the python [logging docs](https://docs.python.org/3/howto/logging.html), they suggest `warnings.warn`:
![image](https://user-images.githubusercontent.com/71127464/120653034-3ccb2d00-c478-11eb-88db-7a7bc75b5d88.png)